src/main/java/org/commcare/formplayer/repo/MediaMetaDataRepo.java

```diff
@@ -2,22 +2,17 @@
 
 import org.commcare.formplayer.objects.MediaMetadataRecord;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.data.jpa.repository.Modifying;
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
-import java.util.Optional;
 
 /**
  * JpaRepository interface for {@link MediaMetadataRecord}
  */
 public interface MediaMetaDataRepo extends JpaRepository<MediaMetadataRecord, String> {
 
-    Optional<MediaMetadataRecord> findByFormSessionId(String formSessionId);
+    List<MediaMetadataRecord> findByFormSession(String formSessionId);
 
-    @Modifying
-    @Transactional
-    @Query("DELETE FROM MediaMetadataRecord WHERE formSession IS NULL")
-    int deleteMetaDataWithoutFormSessionId();
+    @Query(value = "SELECT * FROM media_meta_data WHERE formsessionid IS NULL", nativeQuery = true)
```

Comment:
we probabaly don't need to specify manual query here ? (Since the source of our errors was something else)

Reply:
yeah, that makes sense to me.